llarp/router/rc_gossiper.cpp

```diff
@@ -0,0 +1,68 @@
+#include <router/rc_gossiper.hpp>
+#include <messages/dht_immediate.hpp>
+#include <dht/messages/gotrouter.hpp>
+
+namespace llarp
+{
+  RCGossiper::RCGossiper()
+      : I_RCGossiper(), m_Filter(RouterContact::UpdateInterval)
```

Comment:
This filter rate should be quite a bit shorter; UpdateInterval is currently 3h55m with RC expiry at 4h; we want gossip to happen quite a bit more frequently so that if, say, a lokinet is temporarily offline and misses one RC gossip broadcast it won't end up expiring the router.  It also means that if a router changes (for example if it moves to a different machine and the IP changes) the lokinet can't get its updated RC out across the network until this timer is hit, i.e. just under 4 hours.

I'd suggest (though maybe these can be tweaked more):
- 30 minutes on the filter.
- 1h55m "normal" rebroadcast time for the RC.
- 5m rebroadcast time on startup until we see our RC accepted by the network (i.e. gossiped back to us).
- stale RC expiry remains at 4h.

That way you have to miss two consecutive RC broadcasts in order to remove a remote router's RC for staleness, and an updated RC from a router reconfiguration can take effect in a little over 30 minutes.

Reply:
Actually we could shorten those times (including the staleness) a bit.  The gossip is serving as a sort of heartbeat to let us know a router is active, so not having received one in a while is a good sign that a router is down for some reason.

Maybe 15min filter/30min rebroadcast/1h5min for stale?

Reply:
Or lokid's uptime numbers:

30min filter / 1h normal broadcast / 2h5m stale time.

Reply:
using lokid's numbers